core/domain/subtopic_page_services.py

```diff
@@ -0,0 +1,195 @@
+            'Unexpected error: received an invalid change list when trying to '
+            'save topic %s: %s' % (subtopic_page.id, change_list))
+    subtopic_page.validate()
+
+    subtopic_page_model = topic_models.SubtopicPageModel.get(
+        subtopic_page.id, strict=False)
+    if subtopic_page_model is None:
+        subtopic_page_model = topic_models.SubtopicPageModel(
+            id=subtopic_page.id)
+    else:
+        if subtopic_page.version > subtopic_page_model.version:
+            raise Exception(
+                'Unexpected error: trying to update version %s of topic '
+                'from version %s. Please reload the page and try again.'
+                % (subtopic_page_model.version, subtopic_page.version))
+        elif subtopic_page.version < subtopic_page_model.version:
+            raise Exception(
+                'Trying to update version %s of topic from version %s, '
+                'which is too old. Please reload the page and try again.'
+                % (subtopic_page_model.version, subtopic_page.version))
+
+    subtopic_page_model.topic_id = subtopic_page.topic_id
+    subtopic_page_model.html_data = subtopic_page.html_data
+    subtopic_page_model.language_code = subtopic_page.language_code
+    change_dicts = [change.to_dict() for change in change_list]
+    subtopic_page_model.commit(committer_id, commit_message, change_dicts)
+    subtopic_page.version += 1
+
+
+def update_html_data(
+        subtopic_pages, subtopic_id, deleted_subtopic_ids, new_html_data):
+    """Updates the html_data fields of the subtopic page with the given id in
+    the list (if present), provided the id is not in deleted_subtopic_page_ids
+    list.
+
+    Args:
+        subtopic_pages: list(SubtopicPage). The list of subtopic pages that are
+            part of the topic being edited.
+        subtopic_id: str. The id of the subtopic page to edit.
+        deleted_subtopic_ids: list(int). The ids of the subtopics whose pages
+            were deleted in the present change_list (i.e those that are not
+            updated in the datastore yet).
+        new_html_data: str. The new html data for the given subtopic id.
+
+    Raises:
+        Exception. The subtopic page doesn't exist.
+
+    Returns:
+        list(SubtopicPage). The modified list of subtopic pages.
+    """
+    if subtopic_id in deleted_subtopic_ids:
+        raise Exception(
+            'The subtopic with id %s doesn\'t exist' % subtopic_id)
+    for subtopic_page in subtopic_pages:
+        if (
+                subtopic_page.id ==
+                subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
+                    subtopic_page.topic_id, subtopic_id)):
+            subtopic_page.update_html_data(new_html_data)
+            return subtopic_pages
+
+    raise Exception(
+        'The subtopic with id %s doesn\'t exist' % subtopic_id)
+
+
+def delete_subtopic_page(
+        committer_id, subtopic_id, topic_id, force_deletion=False):
```

Comment:
Swap topic_id and subtopic_id: the former is the parent of the latter and should therefore come first.

Reply:
Done.